### PR TITLE
docs: add changelog with notes for 1.6.0

### LINKED ISSUE #724: Prepare release 1.6.0
### What needs to get done

Do the work to cut the next "feature" release, 1.6.0. This includes:

- Merge whatever is in-flight/review and can sensibly be merged
- Releasing new minor versions of craft-cli and craft-application
- Preparing a good user-oriented changelog describing the relevant changes in Rockcraft and and the libraries since 1.5.3
- Announce the 1.6.0 candidate and do a call for testing

### Why it needs to get done

Our last release was back in July and a *lot* has changed in the craft libraries in the meantime.

### PR DESCRIPTION
**Pull request by tigarmo**
Fixes #724

- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: docs/reference/changelog.rst
Comment (tigarmo) [inline docs/reference/changelog.rst:1]: https://canonical-ubuntu-documentation-library--734.com.readthedocs.build/rockcraft/en/734/reference/changelog/

File: docs/reference/changelog.rst [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+For a complete list of commits, check out the `1.6.0`_ release on GitHub.
+
+
+.. _FastAPI:        https://fastapi.tiangolo.com
+.. _Poetry:         https://python-poetry.org
+
+.. _1.6.0:          https://github.com/canonical/rockcraft/releases/tag/1.6.0
+
+..
```
Comment (medubelko) [inline docs/reference/changelog.rst:107]: I can think of arguments for putting it in either place, but FYI in Snapcraft we have the template at the top of the file.

File: docs/reference/changelog.rst [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+1.6.0 (2024-Oct-17)
+-------------------
+
+Core
+====
+
+- **Note**: This release of Rockcraft updates Pydantic, the internal library
+  used to process ``rockcraft.yaml`` files, to a new major version. This change
+  should not have any user-visible consequences but please report any issues at
```
Comment (medubelko) [inline docs/reference/changelog.rst:16]:
```suggestion
- Rockcraft 1.6.0 updates Pydantic, the internal library
  used to process ``rockcraft.yaml`` files, to a new major version. This change
  should not have any user-visible consequences, but please report any issues at
```

**Review by medubelko — CHANGES_REQUESTED** [review]

File: docs/conf.py [hunk 3]
```diff
@@ -97,6 +97,9 @@
     "https://github.com/canonical/pebble#layer-specification",
     "https://juju.is/cloud-native-kubernetes-usage-report-2021#selection-criteria-for-container-images",
     "https://matrix.to/#/#rocks:ubuntu.com",
+    # Ignore changelog links to Rockcraft releases, because the changelog entries
```
Comment (lengau) [inline docs/conf.py:100]: <3

File: docs/reference/changelog.rst [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+  :literalref:`https://github.com/canonical/rockcraft/issues`.
```
Comment (lengau) [inline docs/reference/changelog.rst:17]:
```suggestion
`on GitHub <https://github.com/canonical/rockcraft/issues>`_.
```

?

File: docs/reference/changelog.rst [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+Init
+====
+
+- Add new ``--profile`` argument options for ``go-framework`` and
+  ``fastapi-framework``.
+
+Documentation
+=============
+
+- Add reference documentation for the new ``poetry`` plugin and the new
```
Comment (lengau) [inline docs/reference/changelog.rst:92]: What about adding `:ref:` links for these?

File: docs/reference/changelog.rst [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+============
+
+- Improve reporting of builds that fail because they match multiple platforms.
+- Improve error messages for invalid ``rockcraft.yaml`` files.
+- The ``pack`` command now supports ``--shell`` and ``--shell-after``, and
+  correctly handles cases where the packing itself fails and the ``--debug``
+  argument is passed.
+- The ``clean`` command now supports the ``--platform`` argument to filter which
+  build environments to clean.
+- Positional arguments are now correctly display on the ``help`` output of
```
Comment (mattculler) [inline docs/reference/changelog.rst:79]:
```suggestion
- Positional arguments are now correctly displayed on the ``help`` output of
```

**Review by mattculler — APPROVED** [review]

File: docs/reference/changelog.rst (hunk 1 above)
Comment (tigarmo) [reply docs/reference/changelog.rst:107]: Right, I "took inspiration" from snapcraft's template. My thinking for putting the template at the bottom is that it's a long preamble that I have to scroll over when updating the document, but I could go either way. If we decide to put the template at the top that's fine with me.

File: docs/reference/changelog.rst [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+Core
+====
+
+- **Note**: This release of Rockcraft updates Pydantic, the internal library
+  used to process ``rockcraft.yaml`` files, to a new major version. This change
+  should not have any user-visible consequences but please report any issues at
+  :literalref:`https://github.com/canonical/rockcraft/issues`.
+- Managed (non-destructive) runs now correctly fail if the build plan is empty.
+  For example, trying to build a project with only ``amd64`` in its
+  ``platforms`` will now properly fail when buildings on a non-``amd64``
```
[the remark this answers is not in the document]
Comment (tigarmo) [reply docs/reference/changelog.rst:20]: I was trying to emphasize the fact that "failing to fail" was a bug here

File: docs/reference/changelog.rst (hunk 1 above)
Comment (lengau) [reply docs/reference/changelog.rst:111]: I never really questioned it, but this gives me a weak preference for the template at the bottom.

**Review by lengau — APPROVED** [review]

**Review by medubelko — APPROVED** [review]

**Pull request merged**